dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresMemoryStore.cs

```diff
@@ -10,34 +10,29 @@
 using Microsoft.SemanticKernel.Memory;
 using Npgsql;
 using Pgvector;
-using Pgvector.Npgsql;
 
 namespace Microsoft.SemanticKernel.Connectors.Memory.Postgres;
 
 /// <summary>
 /// An implementation of <see cref="IMemoryStore"/> backed by a Postgres database with pgvector extension.
 /// </summary>
-public class PostgresMemoryStore : IMemoryStore, IDisposable
+/// <remarks>The embedded data is saved to the Postgres database specified in the constructor.
+/// Similarity search capability is provided through the pgvector extension. Use Postgres's "Table" to implement "Collection".
+/// </remarks>
+public sealed class PostgresMemoryStore : IMemoryStore
 {
+    internal const string DefaultSchema = "public";
+
     /// <summary>
-    /// Connect a Postgres database
+    /// Initializes a new instance of the <see cref="PostgresMemoryStore"/> class.
     /// </summary>
-    /// <param name="connectionString">Database connection string. If table does not exist, it will be created.</param>
-    /// <param name="vectorSize">Embedding vector size</param>
-    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    public static async Task<PostgresMemoryStore> ConnectAsync(string connectionString, int vectorSize,
-        CancellationToken cancellationToken = default)
+    /// <param name="dataSource">Postgres data source.</param>
+    /// <param name="vectorSize">Embedding vector size.</param>
+    /// <param name="schema">Database schema of collection tables. The default value is "public".</param>
+    public PostgresMemoryStore(NpgsqlDataSource dataSource, int vectorSize, string schema = DefaultSchema)
     {
-        var dataSourceBuilder = new NpgsqlDataSourceBuilder(connectionString);
-        // Use pgvector
-        dataSourceBuilder.UseVector();
-
-        var memoryStore = new PostgresMemoryStore(dataSourceBuilder.Build());
-        using NpgsqlConnection dbConnection = await memoryStore._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
-        await memoryStore._dbConnector.CreatePgVectorExtensionAsync(dbConnection, cancellationToken).ConfigureAwait(false);
-        await memoryStore._dbConnector.CreateTableAsync(dbConnection, vectorSize, cancellationToken).ConfigureAwait(false);
-        await memoryStore._dbConnector.CreateIndexAsync(dbConnection, cancellationToken).ConfigureAwait(false);
-        return memoryStore;
+        this._dataSource = dataSource;
+        this._dbConnector = new Database(schema, vectorSize);
```

Comment:
We should also provide a way for users to inject their own implementation of `Database`. Please do the following:
1. As mentioned in previous comment, rename `Database` to `PostgresDbClient` (or `PostgresDatabaseClient`).
2. Create interface based on previous step (e.g. `IPostgresDbClient`) and define all methods which are needed for `PostgresMemoryStore` for interaction.
3. Create new constructor overload for `PostgresMemoryStore` and allow to inject new interface.

Advantages:
1. Users will be able to inject their own implementation or use the default one in case of already existing constructor.
2. It will be possible to unit test `PostgresMemoryStore` with mocked `IPostgresDbClient` interface.

Please check `ChromaMemoryStore` implementation for reference.